corehq/warehouse/transforms/sql/application_dim.sql

```diff
@@ -25,3 +25,12 @@ SELECT
     now(),
     '{{ batch_id }}'
 FROM {{ application_staging }}
+ON CONFLICT (application_id) DO UPDATE
+SET application_id = EXCLUDED.application_id
```

Comment:
we probably don't need to update application id since we already know it matches